attools/src/SwiftBuildTool.swift

```diff
@@ -220,106 +380,28 @@ final class ATllbuild : Tool {
-            #if os(OSX)
-                compileOptions.appendContentsOf(["-F", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/"])
-                linkOptions.appendContentsOf(["-F", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/", "-target", "x86_64-apple-macosx10.11", "-Xlinker", "-rpath", "-Xlinker", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/", "-Xlinker", "-bundle"])
-            #endif
-        }
-        if task["xctestStrict"]?.bool == true {
-            #if os(OSX)
-            //inject XCTestCaseProvider.swift
-            var xcTestCaseProviderPath = "/tmp/XXXXXXX"
-            var template = xcTestCaseProviderPath.cStringUsingEncoding(NSUTF8StringEncoding)!
-            xcTestCaseProviderPath = String(CString: mkdtemp(&template), encoding: NSUTF8StringEncoding)!
-            xcTestCaseProviderPath += "/XCTestCaseProvider.swift"
-            try! ATllbuild.xcTestCaseProvider.writeToFile(xcTestCaseProviderPath, atomically: false, encoding: NSUTF8StringEncoding)
-            sources.append(xcTestCaseProviderPath)
-            #endif
-        }
-
-        guard let name = task["name"]?.string else { fatalError("No name for atllbuild task") }
-        
-        let bootstrapOnly: Bool
-
-        if task["bootstrapOnly"]?.bool == true {
-            bootstrapOnly = true
-        }
-        else {
-            bootstrapOnly = false
-        }
-        
-        let sdk: Bool
-        if task["linkSDK"]?.bool == false {
-            sdk = false
-        }
-        else { sdk = true }
-        
         let llbuildyamlpath : String
-
-        if let value = task["llbuildyaml"]?.string {
+        if let value = task[SwiftBuildToolConfig.Keys.ConfigFile]?.string {
             llbuildyamlpath = value
         }
         else {
             llbuildyamlpath = workDirectory + "llbuild.yaml"
         }
-
-        let swiftCPath: String
-        if let c = task["swiftCPath"]?.string {
-            swiftCPath = c
-        }
-        else {
-            swiftCPath = SwiftCPath
-        }
-        
-        let yaml = llbuildyaml(sources, workdir: workDirectory, modulename: name, linkSDK: sdk, compileOptions: compileOptions, linkOptions: linkOptions, outputType: outputType, linkWithProduct: linkWithProduct, swiftCPath: swiftCPath)
-        let _ = try? yaml.writeToFile(llbuildyamlpath, atomically: false, encoding: NSUTF8StringEncoding)
-        if bootstrapOnly { return }
         
-        //SR-566
         let cmd = "\(SwiftBuildToolpath) -f \(llbuildyamlpath)"
         if system(cmd) != 0 {
             fatalError(cmd)
         }
     }
-}
\ No newline at end of file
+}
+
+/**
+ * This is a friendly front-end wrapper for both the `SwiftBuildToolBuild`
+ * and `SwiftBuildToolConfig` tools.
+ */
+final class SwiftBuildTool: Tool {
+    func run(task: Task) {
```

Comment:
We should probably warn the user in this situation:

1.  `xcteststrict = true`
2.  Is not using `SwiftBuidlToolConfig`

This is because when `xcteststrict` = true, we create a file in a temporary directory and inject it into the build phase, so using `llbuild-config` and trying to distribute the `.swift-build` probably doesn't do what you want